src/features/Folding.ts

```diff
@@ -252,9 +252,18 @@ export class FoldingProvider implements vscode.FoldingRangeProvider {
             return 0;
         });
 
-        // Convert the matched token list into a FoldingRange[]
         const foldingRanges = [];
-        foldableRegions.forEach((item) => { foldingRanges.push(item.toFoldingRange()); });
+        foldableRegions
+            // Use a filter to remove regions which have the same starting line number as the previous
+            // region. This removes duplicate or overlapping ranges. The sort function above determines
+            // which region will be used in the event of duplicates
+            .filter((item, index, src) => {
+                // The first element in the array can never be a duplicate
+                if (index === 0) { return true; }
+                return (item.startline !== src[index - 1].startline);
+            })
+            // Convert the matched token list into a FoldingRange[]
+            .forEach((item) => { foldingRanges.push(item.toFoldingRange()); });
```

Comment:
True. I can fix that.

Reply:
Fixed

Reply:
Oh! I liked it with the filter

Reply:
So did I, but hey, Tyler got in first :-)

Reply:
Ok fineeee what about:

```
const foldingRanges = foldableRegions
    .filter((item, index) => index === 0 || item.startline !== src[index - 1].startline)
    .map((item) => item.toFoldingRange());
```

:)

Reply:
Could probably reduce further
``` typescript
return foldableRegions
    .filter((item, index) => index === 0 || item.startline !== src[index - 1].startline)
    .map((item) => item.toFoldingRange());
```

Reply:
Given the previous PR about "avoiding" functional style and use imperative I'm happy to remove `.filter`

Reply:
I know it sounds capricious, but I like the functional style and I think it applies very well here. I'm also aware that V8 is doing its best to make it all efficient. My concern last time was that I was finding it hard to pick apart the behaviour, but I was looking at that particular code again recently and thinking it might be better implemented as a filter.

Reply:
I'm fine with whatever @glennsarti would like to do in response to @rjmholt :)

Reply:
Fixed.